sensor/kubernetes/enforcer/cronjob/suspend.go

```diff
@@ -0,0 +1,62 @@
+package cronjob
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/branding"
+	kubernetesPkg "github.com/stackrox/rox/pkg/kubernetes"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/sensor/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	batchV1      = "batch/v1"
+	batchBeta1v1 = "batch/beta1v1"
+)
+
+// Suspend suspends the cron job
+func Suspend(ctx context.Context, client kubernetes.Interface, deploymentInfo *central.DeploymentEnforcement) (err error) {
+	depName := deploymentInfo.GetDeploymentName()
+	depType := deploymentInfo.GetDeploymentType()
+	forcePatch := true
+
+	if ok, apiErr := utils.HasAPI(client, batchV1, kubernetesPkg.CronJob); ok && apiErr == nil {
+		patch := fmt.Sprintf("{\"metadata\": {\"name\": %s}, \"kind\": %s, \"apiVersion\": %s, \"spec\": {\"suspend\": true}}", depName, depType, batchV1)
```

Comment:
It is concerning that any characters in `depName` that have special meaning for JSON will remain unescaped. I think, using `Unstructured` (`import "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"`) and serializing that to JSON should be safer choice, although I have not tried that myself.

Reply:
changed to %q

Reply:
I believe `%q` does not escape all characters that matter for JSON (e.g. forward slash) and so it's not a proper replacement for JSON escaping. We can argue whether a forward slash or some other sequence is legit in `depName` in the first place, but why not use JSON escaping in the first place?